docker/Dockerfile.amd

```diff
@@ -45,11 +63,52 @@ ENV PATH=${PATH}:/opt/openmpi/bin:$AOMP/bin
 ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/opt/openmpi/lib:$AOMP/lib
 ENV OMPI_CC=$AOMP/bin/clang
 
-# Devito env
+# Install Omnitrace, an AMDResearch profiling tool akin to Nvidia's Nsight-systems
+ENV OMNITRACE_DIR=/opt/omnitrace
+RUN wget https://github.com/AMDResearch/omnitrace/releases/download/v1.7.3/omnitrace-1.7.3-ubuntu-20.04-ROCm-50100-PAPI-OMPT-Python3.sh -q && \
+    chmod +x omnitrace-1.7.3-ubuntu-20.04-ROCm-50100-PAPI-OMPT-Python3.sh && \
+    mkdir -p ${OMNITRACE_DIR} && \
+    ./omnitrace-1.7.3-ubuntu-20.04-ROCm-50100-PAPI-OMPT-Python3.sh --prefix=${OMNITRACE_DIR} --exclude-subdir --skip-license && \
+    rm omnitrace-1.7.3-ubuntu-20.04-ROCm-50100-PAPI-OMPT-Python3.sh
+
+# Install Omniperf, an AMDResearch profiling tool akin to Nvidia's Nsight-compute
+ENV OMNIPERF_DIR=/opt/omniperf
+RUN wget https://github.com/AMDResearch/omniperf/releases/download/v1.0.4/omniperf-1.0.4.tar.gz -q
+RUN tar xfz omniperf-1.0.4.tar.gz && \
+    cd omniperf-1.0.4/ && \
+    mkdir -p ${OMNIPERF_DIR} && \
+    python3 -m pip install -t ${OMNIPERF_DIR}/python-libs -r requirements.txt && \
+    mkdir build && cd build/ && \
+    cmake -DCMAKE_INSTALL_PREFIX=${OMNIPERF_DIR}/1.0.4 -DPYTHON_DEPS=${OMNIPERF_DIR}/python-libs -DMOD_INSTALL_PATH=${OMNIPERF_DIR}/modulefiles .. && \
+    make install && \
+    rm -rf omniperf-1.0.4.tar.gz omniperf-1.0.4/
+ENV PYTHONPATH ${OMNIPERF_DIR}/python-libs:${PYTHONPATH}
+
+# This will only trigger if arch is aomp since the final stage depends on it
```

Comment:
arch or base?

Reply:
Did you see Dockerfile.nvidia?

Reply:
I *think* I should have done the same thing as there but please correct me if I've made a mistake. Testing locally seems fine.

Reply:
arch is correct. This will only be triggered when the final image, defined by arch, is aomp

Reply:
Yeah sorry, I literally thought I had deleted this comment as I realised it directly